backend/test/app_configuration/test_update_configuration.py

```diff
@@ -0,0 +1,155 @@
+                        "hpo-job": "true",
+                        "model": "true",
+                        "notebook-instance": "false",
+                        "training-job": "true",
+                        "transform-job": "true",
+                    },
+                    "ProjectCreation": {"AdminOnly": "true", "AllowedGroups": ["Justice League", "Avengers", "TMNT"]},
+                    "EMRConfig": {
+                        "cluster-sizes": [
+                            {"name": "Small", "size": 3, "master-type": "m5.xlarge", "core-type": "m5.xlarge"},
+                            {"name": "Medium", "size": 5, "master-type": "m5.xlarge", "core-type": "m5.xlarge"},
+                            {"name": "Large", "size": 7, "master-type": "m5.xlarge", "core-type": "p3.8xlarge"},
+                        ],
+                        "auto-scaling": {
+                            "min-instances": 2,
+                            "max-instances": 15,
+                            "scale-out": {"increment": 1, "percentage-mem-available": 15, "eval-periods": 1, "cooldown": 300},
+                            "scale-in": {"increment": -1, "percentage-mem-available": 75, "eval-periods": 1, "cooldown": 300},
+                        },
+                        "applications": [
+                            {"Name": "Hadoop"},
+                            {"Name": "Spark"},
+                            {"Name": "Ganglia"},
+                            {"Name": "Hive"},
+                            {"Name": "Tez"},
+                            {"Name": "Presto"},
+                            {"Name": "Livy"},
+                        ],
+                    },
+                    "SystemBanner": {"Enabled": "true", "TextColor": "Red", "BackgroundColor": "White", "Text": "Jeff Bezos"},
+                },
+            }
+        ),
+        "requestContext": {"authorizer": {"principalId": "jdoe"}},
+    }
+
+
+@pytest.mark.parametrize(
+    "config_scope",
+    [
+        ("global"),
+        ("project_name"),
+    ],
+    ids=[
+        "update_config_app",
+        "update_config_project",
+    ],
+)
+@mock.patch("ml_space_lambda.app_configuration.lambda_functions.app_configuration_dao")
+def test_update_config_success(mock_app_config_dao, config_scope: str):
+    version_id = 1
+    mock_event = generate_event(config_scope, version_id)
+    mock_app_config_dao.create.return_value = None
+
+    # Add 1 to version ID as it's incremented as part of the update
+    success_response = f"Successfully updated configuration for {config_scope}, version {version_id+1}."
+    expected_response = generate_html_response(200, success_response)
+
+    assert lambda_handler(mock_event, mock_context) == expected_response
+
+
+@mock.patch("ml_space_lambda.app_configuration.lambda_functions.app_configuration_dao")
+def test_update_config_outdated(mock_app_config_dao):
+    version_id = 1
+    mock_event = generate_event("global", version_id)
```

Comment:
Add a test for project configs since they have the same restriction. I know they work the same but if anything changes we'll want to catch the regression.